src/init/VarResolver.ts

```diff
@@ -0,0 +1,139 @@
+import yargs from 'yargs';
+import { AsyncHandler } from '../util/handlers/AsyncHandler';
+import { ensureTrailingSlash, modulePathPlaceholder, resolveAssetPath } from '../util/PathUtil';
+
+const defaultConfig = `${modulePathPlaceholder}config/default.json`;
+const defaultVarConfig = `${modulePathPlaceholder}config/app-setup/vars.json`;
+
+/**
+ * CLI options needed for performing meta-process of app initialization.
+ * These options doesn't contribute to components-js vars normally.
+ */
+export const baseYargsOptions: Record<string, yargs.Options> = {
+  config: { type: 'string', alias: 'c', default: defaultConfig, requiresArg: true },
+  loggingLevel: { type: 'string', alias: 'l', default: 'info', requiresArg: true },
+  mainModulePath: { type: 'string', alias: 'm', requiresArg: true },
+  varConfig: { type: 'string', alias: 'v', default: defaultVarConfig, requiresArg: true },
+};
```

Comment:
```suggestion
export const BASE_YARGS_OPTIONS: Record<string, yargs.Options> = {
  config: { type: 'string', alias: 'c', default: defaultConfig, requiresArg: true },
  loggingLevel: { type: 'string', alias: 'l', default: 'info', requiresArg: true },
  mainModulePath: { type: 'string', alias: 'm', requiresArg: true },
  varConfig: { type: 'string', alias: 'v', default: defaultVarConfig, requiresArg: true },
} as const;
```

So far we have used caps for exported non-function variables. The `as const` makes it so type errors are thrown if another class tries to edit the contents.